src/lib/Preview.js

```diff
@@ -488,31 +484,40 @@ class Preview extends EventEmitter {
      * @return {void}
      */
     download() {
-        const { apiHost, queryParams } = this.options;
-
+        const downloadErrorMsg = __('notification_cannot_download');
         if (!canDownload(this.file, this.options)) {
+            this.ui.showNotification(downloadErrorMsg);
             return;
         }
 
-        // Append optional query params
-        const downloadUrl = appendQueryParams(getDownloadURL(this.file.id, apiHost), queryParams);
-        get(downloadUrl, this.getRequestHeaders()).then((data) => {
-            const defaultDownloadUrl = replaceDownloadHostWithDefault(data.download_url);
-            if (isDownloadHostBlocked() || !isCustomDownloadHost(data.download_url)) {
-                // If we know the host is blocked, or we are already using the default,
-                // use the default.
-                openUrlInsideIframe(defaultDownloadUrl);
-            } else {
-                // Try the custom host, then check reachability
-                openUrlInsideIframe(data.download_url);
-                setDownloadReachability(data.download_url).then((isBlocked) => {
-                    if (isBlocked) {
-                        // If download is unreachable, try again with default
-                        openUrlInsideIframe(defaultDownloadUrl);
-                    }
-                });
+        // Make sure to append any optional query params to requests
+        const { apiHost, queryParams } = this.options;
+
+        // If we should download the watermarked representation of the file, generate the representation URL, force
+        // the correct content disposition, and download
+        if (shouldDownloadWM(this.file, this.options)) {
+            const contentUrlTemplate = getProp(this.viewer.getRepresentation(), 'content.url_template');
+            if (!contentUrlTemplate) {
+                this.ui.showNotification(downloadErrorMsg);
+                return;
             }
-        });
+
+            queryParams.response_content_disposition_type = 'attachment';
```

Comment:
What does setting this to attachment accomplish?

Reply:
You had this in your original PR to force [Content Disposition](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition) to `attachment` instead of the default (`inline`?). I assume this came out of a conversation between you and Albert.

Reply:
If you decide to keep this line, I think you should do an object spread / Object.assign for this. If you add this here, you will be modifying the options the user passes in, and all further uses of ```this.options.queryParams``` will have ```response_content_disposition_type``` which may not be intended

Reply:
@tonyjin yeah I think that came from him as a way to force the browser to treat it like a DL. We should experiment if we can replace opening in the iframe with that?

Reply:
I'll be removing `response_content_disposition_type` since we don't need it, but I'll definitely be using `Object.assign` other places instead of directly modifying `queryParams`. Thanks for the reminder @DanDeMicco.